M365/Search-MailboxExtendedProperty.ps1

```diff
@@ -0,0 +1,138 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+#Requires -Modules @{ ModuleName="ExchangeOnlineManagement"; RequiredVersion="3.4.0" }
+#Requires -Modules @{ ModuleName="Microsoft.Graph.Users"; RequiredVersion="2.24.0" }
+#Requires -Modules @{ ModuleName="Microsoft.Graph.Mail"; RequiredVersion="2.24.0" }
+
+<#
+.SYNOPSIS
+    Searches for mailbox items with a specified extended property (aka named property).
+
+.DESCRIPTION
+    For each of the specified mailbox extended properties, this script searches for mailbox items that have the property set.
+    It returns a list of mailbox items and information about the item and the extended property.
+    The list of mailbox items returned can be used to further process the items, such as removing the extended property.
+    There are some limitations: the search is limited to messages (extended properties can exist on folder, contact, calendar instances etc), single value extended properties (not multi-value), and the property value must be a non-null string.
+
+.PARAMETER MailboxExtendedProperty
+    One of more mailbox extended properties to search for in the mailbox, returned by Get-MailboxExtendedProperty.
+
+.EXAMPLE
+    $mailboxExtendedProperty = Get-MailboxExtendedProperty -Identity fred@contoso.com | Where-Object { $_.PropertyName -like '*Some Pattern*' }
+    $messagesWithExtendedProperty = .\Search-MailboxExtendedProperty.ps1 -MailboxExtendedProperty $mailboxExtendedProperty
+#>
+param(
+    [Parameter(Mandatory = $true, Position = 0)]
+    $MailboxExtendedProperty
```

Comment:
After running `$mailboxExtendedProperty = Get-MailboxExtendedProperty -Identity fred@contoso.com | Where-Object { $_.PropertyName -like '*Some Pattern*' }` what is the type of `$mailboxExtendedProperty`? We should either include that type, if it is specific to a class as a parameter, or we should validate the information passed.

Reply:
Added parameter Type validation